### DISCUSSION TITLE
Azure-pipeline-tasks extractVersion

### DISCUSSION THREAD

**Discussion by charlesRollandy** [renovatebot/renovate#21265] [Request Help] [answered]
### How are you running Renovate?

Self-hosting with Docker Image

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest docker image

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to update version of azure pipeline tasks but the number replaced always contain minor and build part:
![image](https://user-images.githubusercontent.com/7666184/229103520-f1148338-5bfe-40d0-8a1c-1d986290d21a.png)
![image](https://user-images.githubusercontent.com/7666184/229103731-1bad5bea-c70f-4ff6-a9d5-6a2a41e80b89.png)

### What you tried so far.

This is how my renovate.json file looks like (I removed the rules not relevant for this datasource):
```js
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":disableDependencyDashboard"
  ],
  "labels": [
    "Dependencies",
    "Renovate"
  ],
  "azureWorkItemId": 1362101,
  "additionalReviewers": [
    "XXXX"
  ],
  "fetchReleaseNotes": false,
  "major": {
    "enabled": false
  },
  "minor": {
    "enabled": true
  },
  "packageRules": [
    {
      "enabled": false,
      "matchDatasources": [
        "azure-pipelines-tasks"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    },
    {
      "enabled": true,
      "matchDatasources": [
        "azure-pipelines-tasks"
      ],
      "groupName": "Azure Pipeline tasks",
      "matchUpdateTypes": [
        "major"
      ],
      "extractVersion": "^(?<version>\\d+)"
    }
  ]
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
remove match update types from last rule

**Reply by charlesRollandy** [reply]
But by default major updates are disabled:
```cs
 "major": {
    "enabled": false
  },
```
if I remove it how will it apply major updates on the datasource `azure-pipelines-tasks`?

**Reply by viceice** [reply]
just don't combine it with extract version, they are not compatible because the update type isn't known when extract version is applied

**Reply by charlesRollandy** [reply]
Thanks it works!
I find the documentation mission any information about those incompatibilities. It would be nice to improve this
